Keen.NetStandard.Test/QueryTests_Integration.cs

```diff
@@ -117,6 +121,1000 @@ class QueryTests_Integration : TestBase
+                null);
+
+            var expectedResults = expectedResponse["result"];
+
+            Assert.AreEqual(expectedResults.Count(), actualResults.Count());
+            foreach (var group in new string[] { "group1", "group2" })
+            {
+                var actualGroupResult = actualResults.Where((result) => result.Group == group).First();
+                var expectedGroupResult = expectedResults.Where((result) => result[queryParameters.GroupBy].Value<string>() == group).First();
+                foreach (var label in queryParameters.Labels)
+                {
+                    // Validate the result is correct
+                    Assert.AreEqual(expectedGroupResult[label].Value<int>(), int.Parse(actualGroupResult.Value[label]));
+                }
+            }
+        }
+
+        [Test]
+        public async Task Query_SimpleMultiAnalysisInterval_Success()
+        {
+            var queryParameters = new MultiAnalysisParameters()
+            {
+                Labels = new string[]
+                {
+                    "first analysis",
+                    "second analysis"
+                },
+                Analyses = new QueryParameters[]
+                {
+                    new QueryParameters(),
+                    new QueryParameters()
+                    {
+                        Analysis = QueryType.Average(),
+                        TargetProperty = "targetProperty"
+                    }
+                },
+                Interval = QueryInterval.Daily()
+            };
+
+            string responseJson = "{\"result\":[" +
+                    "{\"timeframe\":{\"start\":\"2017-10-14T00:00:00.000Z\",\"end\":\"2017-10-15T00:00:00.000Z\"}," +
+                    "\"value\":{" +
+                        $"\"{queryParameters.Labels[0]}\":12345,\"{queryParameters.Labels[1]}\":54321}}" +
+                    "}," +
+                    "{\"timeframe\":{\"start\":\"2017-10-15T00:00:00.000Z\",\"end\":\"2017-10-16T00:00:00.000Z\"}," +
+                    "\"value\":{" +
+                        $"\"{queryParameters.Labels[0]}\":123,\"{queryParameters.Labels[1]}\":321}}" +
+                    "}" +
+                "]}";
+
+            var expectedResponse = JObject.Parse(responseJson);
+
+            var client = CreateQueryTestKeenClient(queryParameters, expectedResponse);
+
+            var actualResults = await client.Queries.MultiAnalysis(
+                queryParameters.EventCollection,
+                queryParameters.GetMultiAnalysisParameters(),
+                timeframe: null,
+                interval: queryParameters.Interval,
+                filters: null,
+                timezone: null);
+
+            var expectedResults = expectedResponse["result"];
+
+            Assert.AreEqual(expectedResults.Count(), actualResults.Count());
+            var actualResultsEnumerator = actualResults.GetEnumerator();
+            foreach (var expectedResult in expectedResults)
```

Comment:
Can `JToken.DeepEquals()` help in these cases so as to not manually delve into the entire JSON structure, or is it too rigid in that we want to ignore some extraneous properties or something?

Reply:
Maybe, but we'll have to write just as much code probably to transform the actual result into something that will actually be comparable. For example, the SDK returns results as strings while the JSON returned is a number.